rust/driver-ipc/src/sync/client.rs

```diff
@@ -35,25 +37,32 @@ impl Client {
         RUNTIME.block_on(self.0.remove_all())
     }
 
-    /// Receive generic reply
-    ///
-    /// If `last` is false, will only receive new messages from the point of calling
-    /// If `last` is true, will receive the the last message received, or if none, blocks until the next one
-    ///
-    /// The reason for the `last` flag is that replies are auto buffered in the background, so if you send a
-    /// request, the reply may be missed
-    pub fn receive_reply(&mut self, last: bool) -> Option<ReplyCommand> {
-        RUNTIME.block_on(self.0.receive_reply(last))
-    }
-
     /// Receive an event. Only new events after calling this are received
     pub fn receive_event(&mut self) -> EventCommand {
-        RUNTIME.block_on(self.0.receive_event())
+        RUNTIME.block_on(async {
+            self.0
+                .receive_events()
+                .next()
+                .await
+                .expect("Stream should never finish")
+        })
+    }
+
+    pub fn add_event_receiver(&mut self, cb: impl Fn(EventCommand) -> bool + Send + 'static) {
+        let mut stream = self.0.receive_events();
+
+        RUNTIME.spawn(async move {
+            while let Some(event) = stream.next().await {
+                if !cb(event) {
```

Comment:
We also don't know how long `cb` will be executed for, so we should execute this on a blocking thread (maybe a pool?)

Reply:
Hmm, good point. Currently the runtime is configured with only one worker thread. It simply does not need more than this. I am not comfortable with allocating more than that, just because it might be that the user blocks the runtime for too long. I much rather prefer to tell the user to not do that. At least for the Dart bindings this does not matter, because it will be dispached asynchronously anyways (aslo, it uses the async api anyways). In case of Python, can we dispatch it asynchronously too? The GIL blocks it from running concurrently anyways.

But it should be just fine, if the runtime is blocked for a few milliseconds (maybe to update some UI). Events won't be dropped, just delayed. And depending on the users use case, it might also be fine, if it is blocked for a few seconds (But the User should at least be aware of that).

Reply:
A comment mentioning it is fine. I think long blocks should be avoided at all costs, but in principle it's not a problem if the user is aware of that. Pretty much any language has the ability to handle this properly as long as the user knows.

Let's add this warning to the doc comment then.